lib/galaxy/tools/evaluation.py

```diff
@@ -494,11 +495,12 @@ def __build_command_line(self):
             raise
         if interpreter:
             # TODO: path munging for cluster/dataset server relocatability
-            executable = command_line.split()[0]
+            command_line_tokens = shlex.split(command_line)
+            executable = command_line_tokens[0]
             tool_dir = os.path.abspath(self.tool.tool_dir)
             abs_executable = os.path.join(tool_dir, executable)
-            command_line = command_line.replace(executable, abs_executable, 1)
-            command_line = interpreter + " " + command_line
+            command_line_tokens[0:1] = [interpreter, abs_executable]
+            command_line = ' '.join(map(shlex.quote, command_line_tokens))
```

Comment:
Can we revert this commit - it breaks tools that use environment variables (?)


```
python2 /mnt/galaxy/data/galaxy/galaxy-toolshed-tools/testtoolshed.g2.bx.psu.edu/repos/peterjc/mira4_9_mirabait/5979f13c2dc5/mira4_9_mirabait/tools/mira4_9/mirabait/./mira_check_version.py '${MIRA4_9}mirabait' 4.9 '&&' '${MIRA4_9}mirabait' -k 32 -n 6 -b
```

is very different than

```
python2 /mnt/galaxy/data/galaxy/galaxy-toolshed-tools/testtoolshed.g2.bx.psu.edu/repos/peterjc/mira4_9_mirabait/5979f13c2dc5/mira4_9_mirabait/tools/mira4_9/mirabait/./mira_check_version.py ${MIRA4_9}mirabait 4.9 && ${MIRA4_9}mirabait -k "25" -n "6" -b
```

shlex quoting all the arguments in this one scenario and not in all of them also makes it hard to track down these problems and leads to fairly inconsistent behavior right?

Reply:
https://github.com/galaxyproject/galaxy/pull/11343